### PR TITLE
[DoctrineBridge] Fixed #14840

### LINKED ISSUE #14840: [DoctrineBridge] Bug in mapping driver config setter
Description
=========

Because of usage of `realpath` in `Symfony\Bridge\Doctrine\DependencyInjection\AbstractDoctrineExtension`, entities from Phar-packages can't be loaded.

Steps to reproduce
===============

 1. Pack whole the framework with entities in Phar-package or just pack only own code into Phar.
 2. Try to clear the cache or somehow use the entities-related features.
 3. Say "hello" to `Doctrine\Common\Persistence\MappingException` with message `File mapping drivers must have a valid directory path, however the given path  seems to be incorrect!`

### PR DESCRIPTION
**Pull request by saksmt**
[DoctrineBridge] Fixed compatibility with entities packed in Phar

| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #14840 |
| License | MIT |
| Doc PR | none |

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php [hunk 1]
```diff
@@ -129,11 +129,15 @@ protected function setMappingDriverAlias($mappingConfig, $mappingName)
...
     protected function setMappingDriverConfig(array $mappingConfig, $mappingName)
     {
-        if (!is_dir($mappingConfig['dir'])) {
+        $mappingDirectory = $mappingConfig['dir'];
+        if (!is_dir($mappingDirectory)) {
             throw new \InvalidArgumentException(sprintf('Invalid Doctrine mapping path given. Cannot load Doctrine mapping/bundle named "%s".', $mappingName));
         }
 
-        $this->drivers[$mappingConfig['type']][$mappingConfig['prefix']] = realpath($mappingConfig['dir']);
+        if (strpos($mappingDirectory, 'phar://') === false) {
```
Comment (Tobion) [inline src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php]: it should only look at the beginning of the str, not anywhere

**Comment by danielrose28** [comment]
I am being sent these random issues via code triage. Is there anything to help with on this one or is it just basically waiting to be merged?

**Comment by saksmt** [comment]
@danielrose28 It just needs to be merged, bug was really small.

**Comment by danielrose28** [comment]
@saksmt Cool. Just wanted to see if there was anything I could do. Thanks.

**Comment by fabpot** [comment]
Thank you @saksmt.

**Pull request closed without merging**